todo_app/data/trello_items.py

```diff
@@ -0,0 +1,104 @@
+from requests.exceptions import RequestException
+
+from todo_app.helpers.trello_helpers import trello_api_request, get_list_id_from_name
+from todo_app.models.data.item import Item
+
+
+valid_item_status_data = {
+    'To Do': {
+        'theme': 'secondary',
+    },
+    'Doing': {
+        'theme': 'info',
+    },
+    'Done': {
+        'theme': 'success',
+    },
+    'Abandoned': {
+        'theme': 'danger',
+    },
+}
+
+
+def get_items():
+    """
+    Fetches all saved items from the board.
+
+    Returns:
+        list: The list of saved items.
+    """
+    try:
+        items = []
+        for status in valid_item_status_data:
+            list_id = get_list_id_from_name(list_name=status)
+            items_in_list = map(Item.from_trello_card, trello_api_request("GET", f"/lists/{list_id}/cards").json())
+            items.extend(items_in_list)
+        return items
+    except RequestException:
+        return None
+
+
+def get_item(id):
+    """
+    Fetches the saved item with the specified ID.
+
+    Args:
+        id: The ID of the item.
+
+    Returns:
+        item: The saved item, or None if no items match the specified ID.
+    """
+    try:
+        card = trello_api_request("GET", f"/cards/{id}").json()
```

Comment:
Being very pedantic, but ideally you should url encode the item id here + below since it's not trusted input

```
import urllib.parse
trello_api_request("DELETE", f"/cards/{urllib.parse.quote(id, safe='')}")
```

In theory someone could say that their card id is "123abc/somethingelse?query=param" and end up making a request to a different path

Reply:
Thanks for flagging this! I tried a different way of removing this vulnerability using Werkzeug's routing to ensure that only 24-digit hex numbers would ever make it this far - interested to know if this would be a sufficient fix